util/net/http_transport_mac.mm

```diff
@@ -289,10 +268,122 @@ - (BOOL)setProperty:(id)property forKey:(NSStreamPropertyKey)key {
+                    return;
+                  }
+                  if (!response) {
+                    LOG(ERROR) << "no response";
+                    sync_rv = false;
+                    dispatch_semaphore_signal(semaphore);
+                    return;
+                  }
+                  auto http_response =
+                      base::mac::ObjCCast<NSHTTPURLResponse>(response);
+                  if (!http_response) {
+                    LOG(ERROR) << "no http_response";
+                    sync_rv = false;
+                    dispatch_semaphore_signal(semaphore);
+                    return;
+                  }
+                  NSInteger http_status = [http_response statusCode];
+                  if (http_status < 200 || http_status > 203) {
+                    LOG(ERROR) << base::StringPrintf(
+                        "HTTP status %ld", implicit_cast<long>(http_status));
+                    sync_rv = false;
+                    dispatch_semaphore_signal(semaphore);
+                    return;
+                  }
+
+                  if (response_body) {
+                    response_body->assign(
+                        static_cast<const char*>([body bytes]), [body length]);
+                  }
+                  sync_rv = true;
+                  dispatch_semaphore_signal(semaphore);
+                }] resume];
+
+    dispatch_semaphore_wait(semaphore, (dispatch_time_t)(10 * NSEC_PER_SEC));
+  }
+  return sync_rv;
+}
+
+bool HTTPTransportMac::ExecuteSynchronously(std::string* response_body) {
+  DCHECK(body_stream());
+
+  @autoreleasepool {
+    NSString* url_ns_string = base::SysUTF8ToNSString(url());
+    NSURL* url = [NSURL URLWithString:url_ns_string];
+    NSMutableURLRequest* request =
+        [NSMutableURLRequest requestWithURL:url
+                                cachePolicy:NSURLRequestUseProtocolCachePolicy
+                            timeoutInterval:timeout()];
+    [request setHTTPMethod:base::SysUTF8ToNSString(method())];
+
+    // If left to its own devices, CFNetwork would build a user-agent string
+    // based on keys in the main bundle’s Info.plist, giving ugly results if
+    // there is no Info.plist. Provide a User-Agent string similar to the one
+    // that CFNetwork would use, but with appropriate values in place of the
+    // Info.plist-derived strings.
+    [request setValue:UserAgentString() forHTTPHeaderField:@"User-Agent"];
+
+    for (const auto& pair : headers()) {
+      [request setValue:base::SysUTF8ToNSString(pair.second)
+          forHTTPHeaderField:base::SysUTF8ToNSString(pair.first)];
+    }
+
+    base::scoped_nsobject<NSInputStream> input_stream(
+        [[CrashpadHTTPBodyStreamTransport alloc]
+            initWithBodyStream:body_stream()]);
+    [request setHTTPBodyStream:input_stream.get()];
+
+    if (http_proxy().empty()) {
+      return ExecuteNormalRequest(request, response_body);
+    } else {
+      std::string proxy = http_proxy() + "/";
```

Comment:
What's the reason for adding a trailing `/` here? I[ see that our sentry.h file defines the full scheme already](https://github.com/getsentry/sentry-native/blob/master/include/sentry.h#LL969C66-L969C76).

Reply:
Yes, but it has very little cost to append the trailing slash at this point, and `CrackURL` would fail with a log otherwise (even if the URL was valid up to that point) and I didn't want to modify its implementation. If the URL is invalid up to the trailing slash, the additional slash won't make it right. If the URL already has a trailing resource slash, the additional slash will be ignored by us anyway. 

So from my POV since there is so little feedback that we can give from that point in the execution, I thought we could be a little forgiving and not introduce a hard verification boundary here because who would it help? Also, counter to the header docs, our `curl` and `winhttp` proxy implementations accept URLs without a trailing slash too). 

As stated in the other comment, we don't have to be overly friendly or tough here, any validation should happen in the API boundary of the Native SDK. In an ideal world, this stuff wouldn't be passed "stringly", but parsed, validated, and placed into a struct once in the Native SDK and then protobufed (or whatever) into the `crashpad_handler`. But I am not going to change this in this PR :-).

Reply:
Sounds reasonable, any validation should be done on the API side - fully agree! 💯